.appveyor.yml

```diff
@@ -49,23 +53,21 @@ test_script:
       "Test Results:`n$($AllFiles | Out-String)"
       # Upload results for AppVeyor test tab
       "Uploading Test Results:`n"
-      Get-ChildItem -Path $ProjectRoot\packages\*\xunit.xml -File | Foreach-Object {
-        $Address = "https://ci.appveyor.com/api/testresults/junit/$($env:APPVEYOR_JOB_ID)"
-        $Source = $_.FullName
-        "$Address $Source"
-        (New-Object 'System.Net.WebClient').UploadFile( $Address, $Source )
+      (New-Object 'System.Net.WebClient').UploadFile("https://ci.appveyor.com/api/testresults/junit/$($env:APPVEYOR_JOB_ID)" , (Resolve-Path .\junit-aggregate.xml) )
+      [xml]$results = Get-Content junit-aggregate.xml
+      $failure = $results.SelectSingleNode("//failure")
+      if ($failure -ne $null) 
+      { 
+          $host.SetShouldExit(1)
+          Write-Host "Forcing build failure due to test failure(s) :'("
       }
-      # Fail script in case of test failures
-      if($NpmTestsExitCode -ne 0) {
-        #$Error | Format-List * -Force
-        exit $NpmTestsExitCode
-      }
-  - npm run coverage:system-tests
```

Comment:
Modified this to run all the tests first and then upload the single aggregate junit result file.

Reply:
This will also fail the build if there's a test failure. I couldn't find an easy way to do this with a junit parser and found this here:
https://help.appveyor.com/discussions/questions/2504-how-to-fail-build-if-uploaded-test-results-have-failures